hr_holidays_public/tests/test_holidays_public.py

```diff
@@ -0,0 +1,215 @@
+
+        holiday3 = self.holiday_model.create({
+            'year': 1994,
+            'country_id': self.env.ref('base.sk').id
+        })
+        self.holiday_model_line.create({
+            'name': 'holiday 6',
+            'date': '1994-11-14',
+            'year_id': holiday3.id
+        })
+
+        holiday1 = self.holiday_model.create({
+            'year': 1995,
+        })
+        for dt in ['1995-10-14', '1995-12-31', '1995-01-01']:
+            self.holiday_model_line.create({
+                'name': 'holiday x',
+                'date': dt,
+                'year_id': holiday1.id
+            })
+
+        self.employee = self.employee_model.create(
+            {
+                'name': 'Employee 1',
+                'address_id': self.env['res.partner'].create(
+                    {
+                        'name': 'Employee 1',
+                        'country_id': self.env.ref('base.sl').id
+                    }
+                ).id
+            }
+        )
+
+    def test_name_get(self):
+        hol = self.holiday_model.create({
+            'year': 1999,
+        })
+        hol_name = hol.name_get()[0]
+        self.assertEqual(hol_name, (hol.id, str(hol.year)))
+
+    def test_duplicate_year_country_fail(self):
+        # ensures that duplicate year cannot be created for the same country
+        with self.assertRaises(ValidationError):
+            self.holiday_model.create({
+                'year': 1995,
+            })
+        with self.assertRaises(ValidationError):
+            self.holiday_model.create({
+                'year': 1994,
+                'country_id': self.env.ref('base.sl').id
+            })
+
+    def test_duplicate_date_state_fail(self):
+        # ensures that duplicate date cannot be created for the same country
+        # state or with state null
+        holiday4 = self.holiday_model.create({
+            'year': 1994,
+            'country_id': self.env.ref('base.us').id
+        })
+        hline = self.holiday_model_line.create({
+            'name': 'holiday x',
+            'date': '1994-11-14',
+            'year_id': holiday4.id
+        })
+        with self.assertRaises(ValidationError):
+            self.holiday_model_line.create({
+                'name': 'holiday x',
+                'date': '1994-11-14',
+                'year_id': holiday4.id
+            })
+        hline.state_ids = [(6, 0, [self.env.ref('base.state_us_35').id])]
+        with self.assertRaises(ValidationError):
+            self.holiday_model_line.create({
+                'name': 'holiday x',
+                'date': '1994-11-14',
+                'year_id': holiday4.id,
+                'state_ids': [(6, 0, [self.env.ref('base.state_us_35').id])]
+            })
+
+    def test_isnot_holiday(self):
+        # ensures that if given a date that is not an holiday it returns none
+        self.assertFalse(self.holiday_model.is_public_holiday(
+            fields.Date.from_string('1995-12-10')
```

Comment:
Here you can use directly

  date (1995, 12, 10)

instead of

  fields.Date.from_string ('1995-12-10').

In this way we guarantee that we will not have problems with the date format of the string passed by parameter, neither now nor in the future.

I suggest the same for all places in this same file where fields.Date.from_string () is used, passing a static string parameter. I think it should only be used when you want to convert to date a date obtained from database

Reply:
👍